src/ledger/LedgerTxn.cpp

```diff
@@ -26,7 +26,44 @@
 
 namespace stellar
 {
-
+namespace
+{
+void
+enforcePreLoadInvariants(LedgerKey const& key, uint32_t ledgerVersion)
+{
+    if (key.type() == TRUSTLINE)
+    {
+        validateTrustLineKey(ledgerVersion, key);
+    }
+    // Note: There are additional invariants that are enforced when
+    // loading offers by account and asset. This type of query is only
+    // performed on the SQL backend, so we do not enforce those invariants here.
+    // See LedgerTxnOfferSQL.cpp::loadOffersByAccountAndAsset.
+}
+void
+enforcePostLoadInvariants(LedgerKey const& key,
```

Comment:
Nit: Rename to something like `enforceBucketListPostLoadInvariants` so it's clear we only use this for BucketListDB loads and not SQL.